metricbeat/module/prometheus/collector/data.go

```diff
@@ -1,83 +1,94 @@
 package collector
 
 import (
-	"strconv"
-	"strings"
-
 	"github.com/elastic/beats/libbeat/common"
+	dto "github.com/prometheus/client_model/go"
+	"math"
+	"strconv"
 )
 
 type PromEvent struct {
 	key       string
-	value     interface{}
+	value     common.MapStr
 	labels    common.MapStr
 	labelHash string
 }
 
-// NewPromEvent creates a prometheus event based on the given string
-func NewPromEvent(line string) PromEvent {
-	// Separate key and value
-	splitPos := strings.LastIndex(line, " ")
-	split := []string{line[:splitPos], line[splitPos+1:]}
-
-	promEvent := PromEvent{
-		key:       split[0],
-		labelHash: "_", // _ represents empty labels
-	}
-
-	// skip entries without a value
-	if split[1] == "NaN" {
-		promEvent.value = nil
-	} else {
-		promEvent.value = convertValue(split[1])
-	}
-
-	// Split key
-	startLabels := strings.Index(line, "{")
-	endLabels := strings.Index(line, "}")
-
-	// Handle labels
-	if startLabels != -1 {
-		// Overwrite key, as key contained labels until now too
-		promEvent.key = line[0:startLabels]
-		promEvent.labelHash = line[startLabels+1 : endLabels]
-		// Extract labels
-		promEvent.labels = extractLabels(promEvent.labelHash)
-	}
-
-	return promEvent
-}
-
-// extractLabels splits up a label string of format handler="alerts",quantile="0.5"
-// into a key / value list
-func extractLabels(labelsString string) common.MapStr {
-
-	keyValuePairs := common.MapStr{}
-
-	// Extract labels
-	labels := strings.Split(labelsString, "\",")
-	for _, label := range labels {
-		keyValue := strings.Split(label, "=")
-		// Remove " from value
-		keyValue[1] = strings.Trim(keyValue[1], "\"")
+func GetPromEventsFromMetricFamily(mf *dto.MetricFamily) []PromEvent {
+	var events []PromEvent
+
+	name := *mf.Name
+	metrics := mf.Metric
+	for _, metric := range metrics {
+		event := PromEvent{
+			key:       name,
+			labelHash: "#",
```

Comment:
this makes sure that all metrics that dont have any tag values gets grouped into a single document. its a carry over from the previous implementation that was there before this change.